### DISCUSSION TITLE
Measuring water tank levels

### DISCUSSION THREAD

**Discussion by Buster14** [jomjol/AI-on-the-edge-device#3260] [General]
### The Feature

Hi, I was looking for the most accurate way to measure rooftop water tank,.. ultrasound, pressure, etc
But then it came to me, what better way than directly put a cam inside water tank, with a proper lighting?
So I remember this project since It's on my to do list for power meter measuring,
But there are some things that not clear to me since my AI knowledge is rather thin..

- Is there sufficient cpu power to differentiate water tank levels with esp32 cam? I was going for POE cam, but there doesn't seem to be any matured AI project for POE cams.. maybe cause most of them are closed source
- Second possible issue that might arise if the esp32cam is placed inside the tank is moist, is there any known water-proof casing for this project? 
- And third, since espcam hardware can be tricky to choose, would something like this work? Wifi would not be ideal since it's placed rooftop and inside the tank, so POE would be more effective..
 https://www.elecrow.com/esp32-stick-poe-a-cam-n16r8.html?srsltid=AfmBOoqPfjGQDGz4c5tnOy1rOsVua3KzjbKHu2zXBqRSOXBcMP9AksgM

**Comment by jomjol** [comment]
Hi @Buster14 ,
this is not the right code to do this. Here you recognize a feature in a rectangluar ROI and train this with a lot of example images.
What you might need is something much simpler: take an image, make a line scan down and determine, where your water surface is. 
Best regards,
  jomjol

**Comment by Buster14** [comment]
I see, thanks for the explanation,
I guess its a bit relieving knowing its simpler, 
Any idea if there's any tool to get me started with this?
I kinda imagine since the cam is placed on the lid, 

the AI or whatever tool needed would differentiate based on the radius of the water surface picture..
e.g larger radius = higher water level

**Comment by caco3** [comment]
@Buster14, I use a VL53L0X Time-Of-Flight (ToF) Laser sensor to measure the fill level of my water tank. I have put it on top of it and it works reliable despite the water surface.

This way is much easier and requires only i2c but no AI.

**Reply by Buster14** [reply]
Thanks for the suggestion, but a friend of mine used similar sensor and the results weren't too accurate due to water fluctuation...
It could also be his sensor placement that affects this,
But with a cam we have more complete idea of what's going on inside the tank, issues like bugs, moss, float sensor condition if there is one can be spotted easier and earlier.
In my country most of the tanks are placed rooftop so live monitoring with a cam is very useful if possible.
Any idea what sort of tool is needed to recognize water diameter ? 
If the result is not feasible I will try a TOF sensor or other solutions..
